### ISSUE TITLE
Why padding with eos_id in transformer example?

### ISSUE THREAD

**Issue by tungloong** [asyml/texar-pytorch#154]
https://github.com/asyml/texar-pytorch/blob/026163887f1bca450f51a4730986a953d940050c/examples/transformer/utils/data_utils.py#L93-L101

I think we should pad the examples with pad_id (0) to `max_len_seq + 1` of batch. Then only modify one value to eos_id (2) after actual length of each example.

It seems that texar-tf did so.
https://github.com/asyml/texar/blob/4c8b0c0dc2f03d8557ed66a7727fa7efa77cfa08/examples/transformer/utils/data_utils.py#L64

```
    x_seqs, y_seqs = zip(*xy_batch)
    x_block = _concat_examples(x_seqs, padding=0)
    y_block = _concat_examples(y_seqs, padding=0)

    # Add EOS
    x_block = np.pad(x_block, ((0, 0), (0, 1)), "constant", constant_values=0)
    for i_batch, seq in enumerate(x_seqs):
        x_block[i_batch, len(seq)] = eos_id

    y_out_block = np.pad(
        y_block, ((0, 0), (0, 1)), "constant", constant_values=0
    )
    for i_batch, seq in enumerate(y_seqs):
        y_out_block[i_batch, len(seq)] = eos_id
```

**Comment by huzecong** [comment]
Thanks for the issue! I think you're right, this is something we missed while porting from TensorFlow. We will fix this and update the results.

**Comment by gpengzhi** [comment]
Resolved in #156
